src/test/java/uk/gov/hmcts/reform/finrem/caseorchestration/handler/solicitorcreatecase/SolicitorCreateContestedSubmittedHandlerTest.java

```diff
@@ -38,38 +42,72 @@ public class SolicitorCreateContestedSubmittedHandlerTest {
+        FinremCallbackRequest callbackRequest = buildFinremCallbackRequest(YesOrNo.YES);
+        GenericAboutToStartOrSubmitCallbackResponse<FinremCaseData> response = handler.handle(callbackRequest, AUTH_TOKEN);
 
-        verify(createCaseService, times(1)).setSupplementaryData(eq(callbackRequest), any());
+        verify(createCaseService, times(1)).setSupplementaryData(callbackRequest, AUTH_TOKEN);
+        verify(assignApplicantSolicitorService, times(1))
+            .setApplicantSolicitor(callbackRequest.getCaseDetails(), AUTH_TOKEN);
+        assertTrue(response.getErrors().isEmpty());
     }
 
-    private FinremCallbackRequest buildFinremCallbackRequest() {
-        FinremCaseData caseData = FinremCaseData.builder().build();
+    @Test
+    void givenACcdCallbackSolicitorCreateContestedCase_WhenHandleWithAppRepresentationError_thenAddSupplementaryAndHandleException() {
+        FinremCallbackRequest callbackRequest = buildFinremCallbackRequest(YesOrNo.YES);
+
+        doThrow(new AssignCaseAccessException("Exception message"))
+            .when(assignApplicantSolicitorService)
+            .setApplicantSolicitor(any(), any());
+
+        GenericAboutToStartOrSubmitCallbackResponse<FinremCaseData> response = handler.handle(callbackRequest, AUTH_TOKEN);
+
+        verify(createCaseService, times(1)).setSupplementaryData(callbackRequest, AUTH_TOKEN);
+        verify(assignApplicantSolicitorService, times(1))
+            .setApplicantSolicitor(callbackRequest.getCaseDetails(), AUTH_TOKEN);
+
+        assertTrue(response.getErrors().contains("Exception message"));
+        assertTrue(response.getErrors()
+            .contains("Failed to assign applicant solicitor to case, "
+                + "please ensure you have selected the correct applicant organisation on case"));
+    }
+
+    @Test
+    void givenACcdCallbackSolicitorCreateContestedCase_WhenHandleWithNoAppRepresentation_thenApplicantSolicitorNotAssigned() {
+        FinremCallbackRequest callbackRequest = buildFinremCallbackRequest(YesOrNo.NO);
+        GenericAboutToStartOrSubmitCallbackResponse<FinremCaseData> response = handler.handle(callbackRequest, AUTH_TOKEN);
+
+        verify(createCaseService, times(1)).setSupplementaryData(callbackRequest, AUTH_TOKEN);
+        verify(assignApplicantSolicitorService, never()).setApplicantSolicitor(any(), any());
+        assertTrue(response.getErrors().isEmpty());
+    }
+
+    private FinremCallbackRequest buildFinremCallbackRequest(YesOrNo isApplicantRepresented) {
+        FinremCaseData caseData = FinremCaseData.builder()
+            .contactDetailsWrapper(ContactDetailsWrapper.builder().applicantRepresented(isApplicantRepresented).build())
+            .build();
         FinremCaseDetails caseDetails = FinremCaseDetails.builder().id(123L).data(caseData).build();
         return FinremCallbackRequest.builder().eventType(EventType.SOLICITOR_CREATE).caseDetails(caseDetails).build();
     }
-
-}
\ No newline at end of file
+}
```

Comment:
NL

Reply:
New line is there, I think your seeing where I fixed the EOL in the file prior to my change.